examples/Settings.ts

```diff
@@ -26,4 +33,13 @@ async function main() {
   console.log(response.toString());
 }
 
-main().catch(console.error);
+withCallbacks(
```

Comment:
@himself65 very cool - seems also like we can chain it, right? 
Then in this file we could do e.g.:

```suggestion
withLLM(new OpenAI({ model: "gpt-4" }),
withCallbacks(
```